tests/modules/macros.rs

```diff
@@ -38,9 +39,10 @@ fn standard_format() {
         "{{ float: 2.4, string: \"hello\", array: [\"testing\", 1, true, [1, 2]], doc: {{ fish: \
          \"in\", a: \"barrel\", !: 1 }}, bool: true, null: null, regexp: /s[ao]d/i, \
          with_wrapped_parens: -20, code: function(x) {{ return x._id; }}, i32: 12, i64: -55, \
-         timestamp: Timestamp(0, 229999444), binary: BinData(5, 0x{}), _id: ObjectId(\"{}\"), \
-         date: Date(\"{}\") }}",
+         timestamp: Timestamp(0, 229999444), binary: BinData(5, 0x{}), encrypted: BinData(6, \
```

Comment:
this diff has made me realize that I think the `Display` implementation for `Binary` is wrong. The `0x` prefix should be on the subtype not the base64 payload. From looking at the implementation, it looks like its manually adding `0x` rather than using the `{:#x}` hexadecimal formatter. While we're here, can you fix that as part of this change?

Reply:
This change is actually being made as part of [RUST-457](https://jira.mongodb.org/browse/RUST-457)